src/beans.sol

```diff
@@ -0,0 +1,176 @@
+    function rely(address usr) public auth note { wards[usr] = 1; }
+    function deny(address usr) public auth note { wards[usr] = 0; }
+    modifier auth { require(wards[msg.sender] == 1); _; }
+    
+    // --- Data ---
+    // https://github.com/makerdao/dsr/blob/master/src/dsr.sol
+    struct Fee {
+        uint debt;
+        uint chi;
+        uint speed; // Accumulation per second
+        uint48 rho; // Last time the rate was accumulated
+    }
+
+    mapping (uint => Fee) public fees;
+    mapping (uint => uint) public pie;
+
+    uint public debt;
+
+    constructor() public {
+        wards[msg.sender] = 1;
+        fees[0].chi = ONE;
+        fees[0].speed = ONE;
+    }
+
+    function file(uint fee, uint speed_) public auth note {
+        require(speed_ != 0);
+        fees[fee].speed = speed_;
+        fees[fee].chi = ONE;
+        fees[fee].rho = uint48(now);
+        drip(fee);
+    }
+
+    // --- Math ---
+    uint256 constant ONE = 10 ** 27;
+    function rpow(uint x, uint n, uint base) internal pure returns (uint z) {
+        assembly {
+            switch x case 0 {switch n case 0 {z := base} default {z := 0}}
+            default {
+                switch mod(n, 2) case 0 { z := base } default { z := x }
+                let half := div(base, 2)  // for rounding.
+                for { n := div(n, 2) } n { n := div(n,2) } {
+                let xx := mul(x, x)
+                if iszero(eq(div(xx, x), x)) { revert(0,0) }
+                let xxRound := add(xx, half)
+                if lt(xxRound, xx) { revert(0,0) }
+                x := div(xxRound, base)
+                if mod(n,2) {
+                    let zx := mul(z, x)
+                    if and(iszero(iszero(x)), iszero(eq(div(zx, x), z))) { revert(0,0) }
+                    let zxRound := add(zx, half)
+                    if lt(zxRound, zx) { revert(0,0) }
+                    z := div(zxRound, base)
+                }
+            }
+            }
+        }
+    }
+
+    function rmul(uint x, uint y) internal pure returns (uint z) {
+        z = mul(x, y) / ONE;
+    }
+
+    function add(uint x, uint y) internal pure returns (uint z) {
+        require((z = x + y) >= x);
+    }
+
+    function sub(uint x, uint y) internal pure returns (uint z) {
+        require((z = x - y) <= x);
+    }
+
+    function mul(uint x, uint y) internal pure returns (uint z) {
+        require(y == 0 || (z = x * y) / y == x);
+    }
+
+    function rdiv(uint x, uint y) internal pure returns (uint z) {
+        z = add(mul(x, ONE), y / 2) / y;
+    }
+
+    function div(uint x, uint y) internal pure returns (uint z) {
+        z = x / y;
+    }
+
+    function initLoanDebt(uint loan, uint fee, uint wad) public auth note {
+        drip(fee);
+        uint pie_ = calcPie(fees[fee].chi, wad);
+        pie[loan] = pie_;
+        incDebt(fee, wad);
+    }
+
+    function decLoanDebt(uint loan, uint fee, uint wad) public auth note {
+        uint pie_ = calcPie(fees[fee].chi, wad);
```

Comment:
same here.

We also need a consistent use of `_`.

In the context of pile we used it for `delta chi's`.